src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -386,9 +510,82 @@ public Builder withToken(ValueProvider<String> token) {
-      checkArgument(token != null, "withToken(token) called with null input.");
       return setToken(ValueProvider.StaticValueProvider.of(token));
     }
+  
+    /**
+     * Method to set the KMS Encryption Key for HEC token.
+     *
+     * @param tokenKmsEncryptionKey KMS Encryption Key for HEC token
+     * @return {@link Builder}
+     */
+    public Builder withTokenKmsEncryptionKey(ValueProvider<String> tokenKmsEncryptionKey) {
+      return setTokenKmsEncryptionKey(tokenKmsEncryptionKey);
+    }
+  
+    /**
+     * Same as {@link Builder#withTokenKmsEncryptionKey(ValueProvider)} but without {@link ValueProvider}.
+     *
+     * @param tokenKmsEncryptionKey KMS Encryption Key for HEC token
+     * @return {@link Builder}
+     */
+    public Builder withTokenKmsEncryptionKey(String tokenKmsEncryptionKey) {
+      return setTokenKmsEncryptionKey(ValueProvider.StaticValueProvider.of(tokenKmsEncryptionKey));
+    }
+  
+    /**
+     * Method to set the Secret Manager Secret Id for HEC token.
+     *
+     * @param tokenSecretId Secret Manager Secret Id for HEC token
+     * @return {@link Builder}
+     */
+    public Builder withTokenSecretId(ValueProvider<String> tokenSecretId) {
+      return setTokenSecretId(tokenSecretId);
+    }
+  
+    /**
+     * Same as {@link Builder#withTokenSecretId(ValueProvider)} but without {@link ValueProvider}.
+     *
+     * @param tokenSecretId Secret Manager Secret Id for HEC token
+     * @return {@link Builder}
+     */
+    public Builder withTokenSecretId(String tokenSecretId) {
+      return setTokenSecretId(ValueProvider.StaticValueProvider.of(tokenSecretId));
+    }
+  
+    /**
+     * Source of the token. One of PLAINTEXT, KMS or SECRET_MANAGER.
+     *
+     * @param tokenSource Source of the token
+     * @return {@link Builder}
+     */
+    public Builder withTokenSource(ValueProvider<String> tokenSource) {
+      checkArgument(tokenSource != null, "withTokenSource(tokenSource) called with null input");
+  
+      if (tokenSource.isAccessible() && !EnumUtils.isValidEnum(SplunkTokenSource.class, tokenSource.get())) {
+        LOG.warn("tokenSource is invalid. Defaulting to {}", UNKNOWN_TOKEN_SOURCE);
+        return setTokenSource(ValueProvider.StaticValueProvider.of(UNKNOWN_TOKEN_SOURCE));
+      }
+      
+      return setTokenSource(tokenSource);
+    }
+  
+    /**
+     * Same as {@link Builder#withTokenSource(ValueProvider)} but without {@link ValueProvider}.
+     *
+     * @param tokenSource Source of the token
+     * @return {@link Builder}
+     */
+    public Builder withTokenSource(String tokenSource) {
+      checkArgument(tokenSource != null, "withTokenSource(tokenSource) called with null input");
+  
+      if (!EnumUtils.isValidEnum(SplunkTokenSource.class, tokenSource)) {
+        LOG.warn("tokenSource is invalid. Defaulting to {}", UNKNOWN_TOKEN_SOURCE);
+        return setTokenSource(ValueProvider.StaticValueProvider.of(UNKNOWN_TOKEN_SOURCE));
+      }
```

Comment:
Remove this block? The same validation will be performed in `withTokenSource(ValueProvider<String> tokenSource)` method, and `StaticValueProvider` always returns `isAccessible() == true`.

Same for other `with*` methods.